test/executable/execute.ts

```diff
@@ -0,0 +1,644 @@
+      }
+    });
+
+    agentica.on("spawn", () => {
+      runSequence();
+    });
+
+    agentica.on("error", (err) => {
+      console.error(`[ERROR] ${err.message}`);
+      reject(err);
+    });
+  });
+}
+
+function runNpmCommand(command: string, cwd: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const proc = spawn("npm", command.split(" "), {
+      cwd,
+      shell: true,
+      stdio: "pipe",
+    });
+
+    let output = "";
+    proc.stdout.on("data", (data) => (output += data.toString()));
+    proc.stderr.on("data", (data) => (output += data.toString()));
+
+    proc.on("close", (code) => {
+      if (code === 0) resolve(output);
+      else
+        reject(new Error(`npm ${command} failed with code ${code}\n${output}`));
+    });
+  });
+}
+
+function extractServiceInfo(
+  connectorName: string,
+  sourceDir: string,
+): { envList: string[]; hasFileManager: boolean } {
+  try {
+    // 1. 경로 설정
+    const connectorPackageDir = path.join(sourceDir, connectorName);
+    const srcDir = path.join(connectorPackageDir, "src");
+    const structuresDir = path.join(srcDir, "structures");
+
+    // 2. structures 디렉토리 존재 확인
+    if (!fs.existsSync(structuresDir)) {
+      console.log(
+        `[${connectorName}] structures 디렉토리가 없습니다: ${structuresDir}. ENV_LIST 및 FileManager 확인 불가.`,
+      );
+      return { envList: [], hasFileManager: false };
+    }
+
+    let envList: string[] = [];
+    let hasFileManager = false;
+    let foundEnvList = false;
+    let alreadyCheckedFileManager = false;
+
+    // 3. structures 디렉토리 내의 .ts 파일 탐색
+    const structureFiles = fs
+      .readdirSync(structuresDir)
+      .filter((file) => file.endsWith(".ts"));
+
+    console.log(
+      `[${connectorName}] Searching in ${structuresDir} for ENV_LIST and IProps...`,
+    );
+
+    for (const file of structureFiles) {
+      // 최적화: 모든 정보를 이미 찾았으면 루프 종료
+      if (foundEnvList && alreadyCheckedFileManager) {
+        console.log(
+          `[${connectorName}] Found both ENV_LIST and confirmed FileManager status. Stopping search.`,
+        );
+        break;
+      }
+
+      const filePath = path.join(structuresDir, file);
+      try {
+        const content = fs.readFileSync(filePath, "utf8");
+
+        // 3.1 ENV_LIST 찾기 (아직 못 찾은 경우)
+        if (!foundEnvList) {
+          const envListMatch = content.match(
+            /export\s+const\s+ENV_LIST\s*=\s*\[([\s\S]*?)\]/,
+          );
+          if (envListMatch) {
+            const envListStr = envListMatch[1];
+            const envVarMatches = envListStr.match(/'([^']+)'|"([^"]+)"/g);
+            if (envVarMatches) {
+              envList = envVarMatches.map((match) =>
+                match.replace(/['"]/g, ""),
+              );
+              console.log(
+                `[${connectorName}] Found ENV_LIST in ${filePath}:`,
+                envList,
+              );
+              foundEnvList = true; // 찾았음을 표시
+            }
+          }
+        }
```

Comment:
imo. in my opinion) In this part, I think you can use the dynamic import to get the `const ENV_LIST`.